src/apis/solverapi.py

```diff
@@ -0,0 +1,83 @@
+"""
+OrderbookAPI for fetching relevant data using the CoW Swap Orderbook API.
+"""
+# pylint: disable=logging-fstring-interpolation
+
+from os import getenv
+from typing import Any, Optional
+import json
+import requests
+from dotenv import load_dotenv
+from src.models import OrderExecution
+from src.helper_functions import get_logger
+from src.constants import (
+    header,
+    REQUEST_TIMEOUT,
+)
+
+
+class SolverAPI:
+    """
+    Class for querying a solver to provide alternative solutions.
+    """
+
+    def __init__(self, url: Optional[str] = None):
+        self.logger = get_logger()
+        if url is None:
+            load_dotenv()
+            self.solver_url = getenv("SOLVER_URL")
+
+    def get_solution(
+        self, auction_instance: dict[str, Any]
+    ) -> Optional[dict[str, Any]]:
+        """
+        Get solution from auction instance.
+        """
+        try:
+            json_solution = requests.post(
+                f"{self.solver_url}"
+                "solve?time_limit=20&use_internal_buffers=false&objective=surplusfeescosts",
+                headers=header,
+                json=auction_instance,
+                timeout=REQUEST_TIMEOUT,
+            )
+            if json_solution.ok:
+                solution = json.loads(json_solution.text)
+            else:
+                return None
+        except requests.RequestException as err:
+            self.logger.warning(
+                "Connection error while computing solution. "
+                f"Auction ID: {auction_instance['metadata']['auction_id']}, error: {err}"
+            )
+            return None
+        return solution
+
+    def get_execution_from_solution(self, solution: dict[str, Any]) -> OrderExecution:
+        """Get the execution of an order from solution.
+        This is only implemented for the case where exactly one order is supposed to be executed.
+        """
+        orders = solution["orders"]
+        if len(orders) == 1:
+            _, order_dict = solution["orders"].popitem()
+            execution = self.get_execution_from_order(order_dict)
+        elif len(orders) == 0:
+            self.logger.debug("Trivial solution found.")
+            execution = OrderExecution(0, 0, 0)
+        else:
+            raise ValueError(f"Unexpected number of orders in solution: {len(orders)}.")
```

Comment:
Maybe to be added as a comment in the code for future reference: although not important at the moment, we might want at some point to run the check with all liquidity orders included in the json sent to the solver, since this can be viewed as "native protocol liquidity" that can give price improvements. Again, no need to change anything at the moment

Reply:
Good point, I will add a comment on it so we do not forget.